### ISSUE TITLE
`DiagrammeR` package is a dependency that is not automatically loaded with `dm`

### ISSUE THREAD

**Issue by DOH-RPS1303** [cynkra/dm#1735]
```
install.packages("dm")

dm <- dm_nycflights13()

dm
```
Fine up to here:
```
── Metadata ──────────────────────────────────────────────────
Tables: `airlines`, `airports`, `flights`, `planes`, `weather`
Columns: 53
Primary keys: 4
Foreign keys: 4
```
Here's where the issue happens:
```
dm %>%
+   dm_draw()
```
Which returns:
```
Error:
! `dm_draw()` needs the 'DiagrammeR' package. Do you need `install.packages("DiagrammeR")` ?
Run `rlang::last_error()` to see where the error occurred.
```
I already solved this problem for myself by installing the package, but given that this is a necessary dependency for a pretty big part of this package and that it's one of the example code snippets, it'd be nice to have DiagrammeR automatically install with dm

```

**Comment by krlmlr** [comment]
Thanks. This is difficult because CRAN limits the number of packages we can import to 20. We're at 17 now. Theoretically, there's room, and I agree that DiagrammeR might be just important enough to warrant inclusion.

The other issue is that it's quite a heavy dependency, but so are many other packages we're already importing.

Would you like to contribute a PR?

**Comment by krlmlr** [comment]
Thanks. I don't follow: `register_pkgdown_methods()` has an effect only if the `IN_PKGDOWN` environment variable is set. How does that break `R CMD build` ?

Good catch about validation. Has DiagrammeR been validated at some point?

**Comment by pawelru** [comment]
> Thanks. I don't follow: register_pkgdown_methods() has an effect only if the IN_PKGDOWN environment variable is set. How does that break R CMD build ?

I am sorry, I confused CI steps. You are definitely right - this is a `pkgdown` step that failed for me - not R CMD BUILD. Nevertheless, my request stayed the same - to allow draw-free usage without aforementioned packages.

> Has DiagrammeR been validated at some point?

Not until now. At least within an internal env that I am working on (maybe it was not requested up until now?).

**Comment by krlmlr** [comment]
This fell through the cracks, I didn't see an action item here. I believe you can use dm without installing DiagrammeR. What fails for you?
